tests/api/api/marketplace/test_marketplace.py

```diff
@@ -304,3 +305,61 @@ def test_marketplace_catalog_apis(
     function_modified_name = item["metadata"]["name"].replace("_", "-")
 
     assert yaml_function_name == function_modified_name
+
+
+def test_marketplace_get_asset_from_default_source(
+    db: Session, client: TestClient, k8s_secrets_mock: tests.api.conftest.K8sSecretsMock
+) -> None:
+    possible_assets = [
+        ("docs", "text/html; charset=utf-8"),
+        ("source", "text/x-python; charset=utf-8"),
+        ("example", "application/octet-stream"),
+        ("function", "application/octet-stream"),
+    ]
+    sources = client.get("marketplace/sources").json()
+    source_name = sources[0]["source"]["metadata"]["name"]
+    catalog = client.get(f"marketplace/sources/{source_name}/items").json()
+    for _ in range(10):
+        item = random.choice(catalog["catalog"])
+        asset_name, expected_content_type = random.choice(possible_assets)
+        response = client.get(
+            f"marketplace/sources/{source_name}/items/{item['metadata']['name']}/assets/{asset_name}"
+        )
+        assert response.status_code == http.HTTPStatus.OK.value
+        assert response.headers["content-type"] == expected_content_type
+
+
+def test_marketplace_get_asset_1(
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
+) -> None:
+    manager = mlrun.api.crud.Marketplace()
+
+    # Adding marketplace source with credentials:
+    credentials = {"secret": "value"}
+
+    source_dict = _generate_source_dict(1, "source", credentials)
+    expected_credentials = {
+        mlrun.api.crud.Marketplace()._generate_credentials_secret_key(
+            "source", "secret"
+        ): credentials["secret"]
+    }
+    source_object = mlrun.api.schemas.MarketplaceSource(**source_dict["source"])
+    manager.add_source(source_object)
+    k8s_secrets_mock.assert_project_secrets(
+        config.marketplace.k8s_secrets_project_name, expected_credentials
+    )
+    # getting asset:
+    catalog = manager.get_source_catalog(source_object)
+    item = catalog.catalog[0]
```

Comment:
Since we're focusing on assets here, it's best to also verify that the `item` has assets in it (yes, I know `get_asset` would fail if it doesn't, but you can do a more exact check to verify it contains exactly what you expect it to have).